File: PagueMais/Product/entities/Product.cs
namespace Entities
{
  public class Product(string Name, float Price)
  {
    public Guid Id { get; init; } = new();
    public string Name { get; set; } = Name;
    public float Price { get; set; } = Price;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
  }
}

File: client/src/types/product.ts
export type Product = {
  id: string;
  name: string;
  price: number;
  createdAt: Date;
  updatedAt?: Date;
};

export interface CreateProductSchema {
  name: string;
  price: number;
}

export interface UpdateProductSchema extends Partial<CreateProductSchema> {}

export interface UseProductState {
  product: Product | null;
  setProduct: (product: Product | null) => void;
  closeDialog: () => void;
  setIsDeleting: (product: Product) => void;
  setIsUpdating: (product: Product) => void;
}
